sway/input/seat.c

```diff
@@ -368,11 +369,23 @@ static void seat_update_capabilities(struct sway_seat *seat) {
 			caps |= WL_SEAT_CAPABILITY_TOUCH;
 			break;
 		case WLR_INPUT_DEVICE_TABLET_TOOL:
+			if (seat->cursor->proximity == WLR_TABLET_TOOL_PROXIMITY_IN) {
+				caps |= WL_SEAT_CAPABILITY_POINTER;
+			}
+			break;
 		case WLR_INPUT_DEVICE_TABLET_PAD:
 			break;
 		}
 	}
 	wlr_seat_set_capabilities(seat->wlr_seat, caps);
+
+	// Hide cursor if seat doesn't have pointer capability
+	if ((caps & WL_SEAT_CAPABILITY_POINTER) == 0) {
+		wlr_cursor_set_image(seat->cursor->cursor, NULL, 0, 0, 0, 0, 0, 0);
+	} else {
+		wlr_xcursor_manager_set_cursor_image(seat->cursor->xcursor_manager,
+			"left_ptr", seat->cursor->cursor);
```

Comment:
I think we already have a function (in `cursor.c`) to update the cursor (and ensure the cursor state is correctly updated): `cursor_set_image`